src/currencyEngineETH.js

```diff
@@ -589,18 +589,27 @@ class EthereumEngine {
     if (this.addressesChecked) {
       return
     }
-    const numTokens = this.walletLocalData.enabledTokens.length
+    const activeTokens: Array<string> = []
+
+    for (const tokenCode of this.walletLocalData.enabledTokens) {
+      const ti = this.getTokenInfo(tokenCode)
+      if (ti) {
+        activeTokens.push(tokenCode)
+      }
+    }
+    const numTokens = activeTokens.length
+
     const perTokenSlice = 1 / numTokens
     let numCompleteStatus = 0
     let totalStatus = 0
-    for (const token of this.walletLocalData.enabledTokens) {
+    for (const token of activeTokens) {
       const status = this.tokenCheckStatus[token]
       totalStatus += status * perTokenSlice
       if (status === 1) {
         numCompleteStatus++
       }
     }
-    if (numCompleteStatus === this.walletLocalData.enabledTokens.length) {
+    if (numCompleteStatus === activeTokens.length) {
```

Comment:
If you already have 'numTokens' why not use it instead of `activeTokens.length` or getting rid of it if you don't need it?

Reply:
i'll make the change